pandas/tests/test_index.py

```diff
@@ -3425,6 +3433,22 @@ def setUp(self):
     def create_index(self):
         return date_range('20130101', periods=5)
 
+    def test_shift(self):
+        # test shift for datetimeIndex and non datetimeIndex
+        # err8083
+
+        drange = self.create_index()
+        result = drange.shift(1)
+        expected = DatetimeIndex(['2013-01-02', '2013-01-03', '2013-01-04', '2013-01-05',
+               '2013-01-06'], freq='D')
+        self.assert_index_equal(result, expected)
+
+        result = drange.shift(3, freq='2D')
+        expected = DatetimeIndex(['2013-01-07', '2013-01-08', '2013-01-09', '2013-01-10',
```

Comment:
can you add negative shift & a 0 shift test as well (which should return identically)